pkg/collector/corechecks/system/cpu_ctx_switches_linux.go

```diff
@@ -0,0 +1,53 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+// +build linux
+
+package system
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func readCtxSwitches(procStatPath string) (ctxSwitches int64, err error) {
+	file, err := os.Open(procStatPath)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		txt := scanner.Text()
+		if strings.HasPrefix(txt, "ctxt") {
+			elemts := strings.Split(txt, " ")
+			ctxSwitches, err = strconv.ParseInt(elemts[1], 10, 64)
+			if err != nil {
+				return 0, err
+			}
+			return ctxSwitches, nil
+		}
+	}
```

Comment:
nit: the error when we fail to parse could be a bit confusing since you only get that a certain integer failed to parse. We could add something like:

```suggestion
	for i := 0; scanner.Scan(); i++ {
		txt := scanner.Text()
		if strings.HasPrefix(txt, "ctxt") {
			elemts := strings.Split(txt, " ")
			ctxSwitches, err = strconv.ParseInt(elemts[1], 10, 64)
			if err != nil {
				return 0, fmt.Errorf("failed to parse line %i from %s: %s, i, procStatPath, err)
			}
			return ctxSwitches, nil
		}
	}
```